src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -319,8 +332,16 @@ public function initialized($id)
         if (isset($this->aliases[$id])) {
             $id = $this->aliases[$id];
         }
+        if (isset($this->services[$id])) {
+            return true;
+        }
+        if (array_key_exists($id, $this->services)) {
+            @trigger_error(sprintf('Checking if a unset "%s" service was initialized is deprecated since Symfony 3.2 and won\'t be supported anymore in Symfony 4.0.', $id), E_USER_DEPRECATED);
```

Comment:
but your deprecation warning is a bad experience:

- unsetting the service will **not** trigger the deprecation (as the key won't exist), so your message is lying
- code calling ``initialized`` or ``has`` has strictly no way to know whether a service was unset or no, so forbidding them to call the method for these ids makes the whole method unusable if they want to avoid deprecation warnings.

There should be no deprecation in this method. We are not deprecating anything in ``initialized()`` in this patch.